### PR TITLE
Recurring event module

### PR DESCRIPTION
**Pull request by AlehBarushka**
With the ability to set a recurring event (every day, on selected days of the week).

### CODE REVIEW

File: module/RecurringEvent.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import { timeValidator } from '../utils/date-utils.js';
+import {
+  daysOfWeekValidator,
+  daysOfWeekСonverter,
+} from '../utils/daysOfWeek-utils.js';
```
Comment (vitali-working) [inline module/RecurringEvent.js]: I think  postfix `-utils` is redundant

File: utils/recurringEvent-utils.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+ * @param {String} event.time - time of recurring event in format '12:00:00'.
+ * @returns {(Boolean|Error)} returns true if the event object is valid, or returns an error if otherwise.
+ * @example
+ * // returns true
+ * recurringEventValidator({title: 'meeting', time: '12:00:00'})
+ * // returns Error
+ * recurringEventValidator({title: 'meeting'})
+ **/
+export const recurringEventValidator = (event) => {
+  if (event?.title && event.time) {
```
Comment (vitali-working) [inline utils/recurringEvent-utils.js]: Do you really need check `time` here?

File: utils/daysOfWeek-utils.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ * @description The function check the validity of the array.
+ * @param {Array.<Srting>} daysOfWeek - array with days of the week.
+ * @returns {(boolean|Error)} returns true if the array is valid or returns an error if otherwise.
+ * @example
+ * // returns true
+ * daysOfWeekValidator(['Sunday', 'Monday', 'Tuesday'])
+ * // returns Error
+ * daysOfWeekValidator('Monday')
+ **/
+export const daysOfWeekValidator = (daysOfWeek) => {
```
Comment (vitali-working) [inline utils/daysOfWeek-utils.js]: A would say it's a bit strange validation. I think you can extend validation and add max length for array(array.length <= 7).

File: utils/date-utils.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+ * @description The function set the time and parse the set date of the recurring event.
+ * @param {string} time - time of recurring event in format '15:00:00'.
+ * @returns {Number} returns the set date in milliseconds.
+ **/
+export const setTime = (time) => {
+  const currentDate = new Date();
+
+  const arrOfSettedTime = time.split(':');
```
Comment (vitali-working) [inline utils/date-utils.js]: I guess is more readable
`const [hours, minutes, seconds] = time.split(':');`

File: utils/daysOfWeek-utils.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ * @param {Array.<Srting>} daysOfWeek - array with days of the week.
+ * @returns {Array.<Number>} returns an array with the days of the week in numerical representation.
+ * @example
+ * // returns [0, 1, 2]
+ * daysOfWeekСonverter(['Sunday', 'Monday', 'Tuesday'])
+ * // returns [1, 2, 0]
+ * daysOfWeekСonverter(['Monday', 'Tuesday', 'Sunday'])
+ **/
+export const daysOfWeekСonverter = (daysOfWeek) => {
+  const days = [
```
Comment (newqwes) [inline utils/daysOfWeek-utils.js]: Move it separately to the constants file

File: utils/daysOfWeek-utils.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    'Sunday',
+    'Monday',
+    'Tuesday',
+    'Wednesday',
+    'Thursday',
+    'Friday',
+    'Saturday',
+  ];
+
+  let arrOfDayNumbers = [];
```
Comment (newqwes) [inline utils/daysOfWeek-utils.js]: Here it is better to using reduce method

File: utils/daysOfWeek-utils.js (hunk 3 above)
Comment (newqwes) [inline utils/daysOfWeek-utils.js]: The name does not match, this is rather a check for an array and not a validator for `daysOfWeekValidator`

File: utils/recurringEvent-utils.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+ * @param {Function} callback - callback function that will be called at the interval.
+ * @param {Number} delay - setTimeout delay in millliseconds.
+ * @param {Number} interval - setInterval delay in millliseconds.
+ * @param {String} eventId - recurringEvent ID.
+ * @returns {Number} returns setTimout ID.
+ * @example
+ * // returns 12
+ * loop(() => console.log('Hello'), 10000, 86400000, 'f9ca-1baa-c970-35b4')
+ **/
+const loop = (callback, delay, interval, eventId) => {
```
Comment (newqwes) [inline utils/recurringEvent-utils.js]: 4 function parameters is a lot, it is better to pass no more than two, the rest if more then an object

File: utils/recurringEvent-utils.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+ * @description The function sets the daily interval with a certain delay.
+ * @param {Function} callback - callback function that will be called at the interval.
+ * @param {String} time - time in format '12:00:00'.
+ * @returns {Object} returns an object with the recurringEvent ID and the setTimout ID.
+ * @example
+ * // returns {eventId: 'f9ca-1baa-c970-35b4', timerId: 12}
+ * dailyLoop(() => console.log('Hello'), '12:00:00')
+ **/
+export const dailyLoop = (callback, time) => {
+  const DAY_IN_MILLISECONDS = 86400000;
```
Comment (newqwes) [inline utils/recurringEvent-utils.js]: Move to variable file

File: utils/recurringEvent-utils.js [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+ * @param {Function} callback - callback function that will be called at the interval.
+ * @param {String} time - time in format '12:00:00'.
+ * @param {Array.<Srting>} daysOfWeek - array with days of the week.
+ * @returns {Object} returns an object with the recurringEvent ID and an array with setTimeout IDs.
+ * @example
+ * // returns {reminderId: 'f9ca-1baa-c970-35b4', timerIDs: [12, 13]}
+ * dailyLoop(() => console.log('Hello'), '12:00:00', ['Monday', 'Tuesday'])
+ **/
+export const daysOfWeekLoop = (callback, time, daysOfweek) => {
+  const WEEK_IN_MILLISECONDS = 604800000;
```
Comment (newqwes) [inline utils/recurringEvent-utils.js]: Move to variable file

File: utils/date-utils.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  } else {
+    throw new Error('Invalid time format');
+  }
+};
+
+/**
+ * @description The function returns the current day of the week.
+ * @returns {Number} returns the current day of the week according to local time, where 0 represents Sunday.
+ **/
+export const getCurrentWeekDay = () => {
```
Comment (newqwes) [inline utils/date-utils.js]: redundant method

File: utils/date-utils.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+  return new Date().getDay();
+};
+
+/**
+ * @description The function set the time and parse the set date of the recurring event.
+ * @param {string} time - time of recurring event in format '15:00:00'.
+ * @returns {Number} returns the set date in milliseconds.
+ **/
+export const setTime = (time) => {
```
Comment (newqwes) [inline utils/date-utils.js]: avoid code duplication. More or less like this:
```
export const getTimeInMillisecond = (time) => {
  const splittedTime = time.split(':')

  const timeInMillisecond = new Date().setHours(...splittedTime)

  return timeInMillisecond
}
```

Comment (AlehBarushka) [reply utils/date-utils.js]: can I pass parameters as strings to the setHours method?
I checked, it works) 
but still, is it right?

**Review by newqwes — APPROVED** [review]

**Pull request merged**